docs/cephfs-snap-clone.md

```diff
@@ -0,0 +1,78 @@
+# Create CephFS snapshot and Clone Volume
+
+- [Create CephFS snapshot and Clone Volume](#create-cephfs-snapshot-and-clone-volume)
+  - [Prerequisite](#prerequisite)
+  - [Snapshot](#snapshot)
+    - [Create snapshotclass](#create-snapshotclass)
+    - [Create snapshot](#create-snapshot)
+  - [Restore snapshot to a new PVC](#restore-snapshot-to-a-new-pvc)
+  - [Clone PVC](#clone-pvc)
+
+## Prerequisite
+
+- For snapshot functionality to be supported for your Kubernetes cluster, the
+  Kubernetes version running in your cluster should be >= v1.17. We also need the
+  `snapshot controller` deployed in your Kubernetes cluster along with `csi-snapshotter`
+  sidecar container.
+  Refer [external-snapshotter](https://github.com/kubernetes-csi/external-snapshotter/)
+  for more information on these sidecar controllers. There should
+  be a `volumesnapshotclass` object present in the cluster
+  for snapshot request to be satisified.
+
+## Snapshot
+
+### Create snapshotclass
+
+```console
+kubectl create -f ../examples/cephfs/snapshotclass.yaml
+```
+
+### Create snapshot
+
+The snapshot is created on/for an existing PVC. You should
+have a PVC in bound state before creating snapshot from it.
+please refer pvc creation [doc](https://github.com/ceph/ceph-csi/blob/master/docs/deploy-cephfs.md)
+for more information on how to create a PVC.
+
+```console
+$ kubectl create -f ../examples/cephfs/snapshot.yaml
+volumesnapshot.snapshot.storage.k8s.io/cephfs-pvc-snapshot created
+```
+
+- Get details about the snapshot
+
+```console
+$ kubectl get volumesnapshot
+NAME                  READYTOUSE   SOURCEPVC       SOURCESNAPSHOTCONTENT   RESTORESIZE   SNAPSHOTCLASS                SNAPSHOTCONTENT                                    CREATIONTIME   AGE
+cephfs-pvc-snapshot   true         csi-cephfs-pvc                          1Gi           csi-cephfsplugin-snapclass   snapcontent-34476204-a14a-4d59-bfbc-2bbba695652c   3s             6s
+```
+
+- Get details about the volumesnapshotcontent
+
+```console
+$ kubectl get volumesnapshotcontent
+NAME                                               READYTOUSE   RESTORESIZE   DELETIONPOLICY   DRIVER                VOLUMESNAPSHOTCLASS          VOLUMESNAPSHOT        AGE
+snapcontent-34476204-a14a-4d59-bfbc-2bbba695652c   true         1073741824    Delete           cephfs.csi.ceph.com   csi-cephfsplugin-snapclass   cephfs-pvc-snapshot   64s
+```
+
+## Restore snapshot to a new PVC
+
+```console
+
+$ kubectl create -f ../examples/cephfs/pvc-restore.yaml
+$ kubectl get pvc
```

Comment:
sorry @humblec I forgot on the last thing all cephfs PVC access mode is RWX in examples

Reply:
@Madhu-1 changing everything to RWX

Reply:
done.